qualtran/bloqs/rotations/phasing_via_cost_function.py

```diff
@@ -154,30 +158,20 @@ def b_phase(self) -> int:
     @cached_property
     def b_grad(self) -> int:
         # Using Equation A7 from https://arxiv.org/abs/2007.07391
-        eq_a7 = int(np.ceil(np.log2((self.gamma_bitsize + 2) * np.pi / self.eps)))
-        # Using Equation 35 from https://arxiv.org/abs/2007.07391
-        eq_35 = self.b_phase + int(np.ceil(np.log2(self.b_phase)))
-        # TODO(#654): Eq A7 will result in a bigger gradient bitsize but blows up the cost
-        #   for doing phase gradient based cost computations significantly (which leads to
-        #   the cost using PhaseOracleZPow to be cheaper). Also, we don't yet have a test that
-        #   fails for Eq 35. The only concern is that value of `eq_35` can be smaller than
-        #   `cost_reg.bitsize`.
-        assert eq_a7 >= eq_35
-        return eq_35
+        return int(np.ceil(np.log2((self.gamma_bitsize + 2) * np.pi / self.eps)))
```

Comment:
What was the resolution here? A7 is correct?

Reply:
See the newly added notebook `qualtran/bloqs/rotations/quantum_variable_rotation.ipynb`. 

Tl;Dr is that both equations were correct and equivalent given the assumptions in the paper. In the general case, A7 is more accurate.